content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/2. Uint.md

```diff
@@ -0,0 +1,41 @@
+---
+title: 2. Uint
+taxonomie:
+  - ib-19.2.Datatype-Primitief-Uint.OI
+  - ib-19.3.Datatype-Primitief-Uint.OI
+---
+
+### Wat is een Uint 
+Een Uint, ofwel unsigned integer, is een datatype die dezelfde ruimte in beslag neemt als de [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/1. Integer|Integer]]. De Uint kan alleen geen negatieve getallen opslaan, maar gebruikt deze ruimte om een twee keer zo groot positief getal op te kunnen slaan (4,294,967,295). 
+
+Voor de rest werkt de Uint hetzelfde als de [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/1. Integer|Integer]].
+
+> [!TIP] Casus
+> Stel je hebt een evenement met waar van tevoren tickets voor worden gekocht. Je wilt bijhouden hoeveel tickets er zijn verkocht.  
+
+### Hoe zit een Uint in elkaar
+De structuur van de Uint werkt hetzelfde als die van de [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/1. Integer|Integer]]. Alleen moet je nu Uint voor het variable neerzetten.
+
+### Codevoorbeeld
+```C#
+uint variabelNaam;
+```
+
+### Hoe gebruik je een Uint
+De Uint wordt vooral gebruikt wanneer het iets representeert wat nooit negatief kan zijn, zoals bij de ticketverkoop.
+
+> [!TIP] Casus
+> Stel je hebt maakt een programma waarbij het aantal verkochte tickets voor een evenement wordt bijgehouden. Het programma kan een Uint gebruiken om het aantal verkochte tickets bij te houden. Bij een aankoop gaat de waarde met 1 omhoog, en bij een refund omlaag, zonder onder nul te komen.
+
+**Mogelijke uitwerking van de casus**
+```C#
+uint count = 0; 
+count = count + 1; // Er is een ticket gekocht.
+Console.WriteLine(count); // count = 1
+count = count - 1; // Er word een ticket gerefund.
```

Comment:
Opmerking: Ah~ Maar, wat als count 0 is en er wordt -1 gedaan (ja I know, gebeurt niet met deze code)? Blijft de uint dan gewoon 0? Of is dit iets om een opdrachtje bij te maken later om dit aan te tonen?

Reply:
Ik heb het in de OI tekst toegevoegd.